training/src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -12,9 +14,10 @@
 
     Optional<Users> findByUsername(String username);
 
+    @Query(value = "SELECT u FROM Users u WHERE UPPER(name) LIKE UPPER(CONCAT(:name,'%')) and (birthdate BETWEEN :birthdateFrom AND :birthdateTo) ")
```

Comment:
It is not wrong, but I think it's not necessary to use concat for `:name` and `%`. Putting `:name%` it's ok.
Also, it should be `%:name%` in order to accept any name containing the provided string,

Reply:
I try to put that. but the next follow error thorw by intellij.

![image](https://user-images.githubusercontent.com/42941989/84951067-3f494100-b0c6-11ea-8d07-a6279f430a57.png)

Caused by: java.lang.IllegalArgumentException: Validation failed for query for method public abstract java.util.Optional wolox.training.repositories.UserRepository.findByBirthdateBetweenAndNameContainingIgnoreCase(java.time.LocalDate,java.time.LocalDate,java.lang.String)!

Reply:
I tried different ways but that's the only one that worked for me

Reply:
another way that worked for me is this:
![image](https://user-images.githubusercontent.com/42941989/84951664-20977a00-b0c7-11ea-93ff-c6eb74d0a8fd.png)

Reply:
Oh, sorry! It worked for me, maybe it's the `UPPER`, nevermind.